test/testUnregisterMachine.js

```diff
@@ -19,7 +23,28 @@ describe('mercurius unregisterMachine', function() {
     .then(() => testUtils.register(mercurius.app, 'machine_1', 'https://android.googleapis.com/gcm/send/someSubscriptionID', ''))
     .then(gotToken => token = gotToken)
     .then(() => testUtils.register(mercurius.app, 'machine_2', 'https://localhost:50005', urlBase64.encode(userPublicKey), token))
-    .then(() => testUtils.register(mercurius.app, 'machine_3', 'https://localhost:50006', urlBase64.encode(userPublicKey), token));
+    .then(() => redis.sadd(token + ':clients', 'someClient'))
+    .then(() => redis.hmset('machine_1:clients', 'someClient', 1));
```

Comment:
Could you add the client without using redis but with a request to the server?
If we introduce too many implementation-specific details in the test, changing the implementation could become a pain.